libraryAPI/cmd/main.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"library"
+	"library/pkg/handler"
+	"library/pkg/repository"
+	"library/pkg/service"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+func main() {
+	if err := InitConfig(); err != nil {
+		logrus.Fatalf("Error on initializing config: %s", err.Error())
+	}
+	repos := repository.NewRepository()
+	services := service.NewService(repos)
+	handlers := handler.NewHandler(services)
+	server := new(library.Server)
+
+	if err := server.Start(viper.GetString("port"), handlers.InitRoutes()); err != nil {
+		logrus.Fatalf("Error ocured while running http server %s", err.Error())
+	}
+}
+
+func InitConfig() error {
```

Comment:
for Go idiomatically using funcs like 'New...'
i would even moved your config logic in a separate package with 'config' name and there you create your just 'New' func
so you will call it from main like
config.New()